source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -517,6 +543,36 @@ void OAuth2Filter::finishGetAccessTokenFlow() {
   config_->stats().oauth_success_.inc();
 }
 
+void OAuth2Filter::setCookie(Http::ResponseHeaderMap& headers, const std::string& key,
+                             const std::string& value) const {
+  headers.addReferenceKey(Http::Headers::get().SetCookie, absl::StrCat(key, "=", value));
+}
+
+void OAuth2Filter::setSplitCookie(Http::ResponseHeaderMap& headers, const std::string& key,
+                                  const std::string& data, size_t maxChunkSize = 4000) const {
```

Comment:
setting default to 4000 should be in the function declaration I think. Rather,  should we make it configuration option?

Reply:
Most modern browsers have 4096 as the limit, 4000 should suffice all cases. If we choose to take this utility function out to Http::Utility then probably we should have the configuration option, wdyt?